samples/Gallery/Pages/TreeView/EditableTreeView.fs

```diff
@@ -0,0 +1,280 @@
+namespace Gallery
+
+open System.Diagnostics
+open System.Runtime.CompilerServices
+open Avalonia.Controls
+open Avalonia.Input
+open Avalonia.Layout
+open Fabulous.Avalonia
+open Fabulous
+
+open type Fabulous.Avalonia.View
+
+module FocusAttributes =
+    /// Allows setting the Focus on an Avalonia.Input.InputElement
+    let Focus =
+        Attributes.defineBool "Focus" (fun oldValueOpt newValueOpt node ->
+            let target = node.Target :?> InputElement
+
+            let rec focusAndCleanUp x y =
+                target.Focus() |> ignore
+                target.AttachedToVisualTree.RemoveHandler(focusAndCleanUp) // to clean up
+
+            if newValueOpt.IsSome && newValueOpt.Value then
+                (* TODO setting the focus on an AutoCompleteBox is broken.
+                    It works for some (probably threading-related) reasion if you hit a magic break point here
+                    or in the focusAndCleanUp handler above. *)
+                Debugger.Break()
+                target.AttachedToVisualTree.AddHandler(focusAndCleanUp))
+
+type FocusModifiers =
+    [<Extension>]
+    /// Sets the Focus on an IFabInputElement if set is true; otherwise does nothing.
+    static member inline focus(this: WidgetBuilder<'msg, #IFabInputElement>, set: bool) =
+        this.AddScalar(FocusAttributes.Focus.WithValue(set))
+
+type EditableNode(name, children) =
+    let mutable _name: string = name
+    let mutable _children: EditableNode list = children
+    let mutable _expanded: bool = false
+
+    //TODO Is a mutable model required or can an updated immutable model be passed to the parent component?
```

Comment:
I do not think is required. Try with a record and see what happens when you bind this to the `TreeViewItem` `IsExpanded`

Reply:
Tried replacing this with a record but it did not work. I guess the mutable model play better with the IsExpanded value binding